runtime/major_gc.c

```diff
@@ -318,7 +318,7 @@ static intnat ephe_mark (intnat budget, uintnat for_cycle,
   mlsize_t size, i;
   caml_domain_state* domain_state = Caml_state;
   int alive_data;
-  intnat marked = 0, made_live = 0;
+  intnat marked = 0, already_marked = 0, made_live = 0;
```

Comment:
It would be nice if: all these variables had better names and types (e.g. `bool mark_data = false; uintnat examined = 0, trivial_data = 0, marked = 0;`)

Reply:
I fixed the one I introduced - I agree the rest of this function could do with some changes but I'll leave them for another time